server/vulnerabilities/nvd/sync/cve_syncer.go

```diff
@@ -488,6 +489,26 @@ func transformVuln(item nvdapi.CVEItem) nvdapi.CVEItem {
 		item.CVE.Configurations[0].Nodes[0].CPEMatch = item.CVE.Configurations[0].Nodes[0].CPEMatch[0:1]
 	}
 
+	// Handle NVD PowerShell version format normalization
+	// 1. Version: 7.5 → 7.5.0
+	// 2. RC notation: preview1 → rc.1
+	if item.CVE.ID != nil && (*item.CVE.ID == "CVE-2025-21171" || *item.CVE.ID == "CVE-2025-30399") {
+		for i := range item.CVE.Configurations {
+			for j := range item.CVE.Configurations[i].Nodes {
+				for k := range item.CVE.Configurations[i].Nodes[j].CPEMatch {
+					cpeMatch := &item.CVE.Configurations[i].Nodes[j].CPEMatch[k]
+					criteria := cpeMatch.Criteria
+
+					criteria = strings.Replace(criteria, "microsoft:powershell:7.5:", "microsoft:powershell:7.5.0:", 1)
+					previewRegex := regexp.MustCompile(`:preview(\d+):`)
+					criteria = previewRegex.ReplaceAllString(criteria, ":rc.$1:")
+
+					cpeMatch.Criteria = criteria
```

Comment:
Let's add these rather than replacing (so, a bit different from what I did above). That way we don't break stuff again if/when we fix PowerShell version parsing or CPE matching in Fleet server.

Would also be helpful to include a diff between pre-fix and -post-fix 2025 CVE feed in the comments, plus QA instructions, since this'll need to be manually QA'd on branch (because this changes more than tests).

Reply:
Good question on how to QA this. I have just been running `VULNCHECK_API_KEY=<...> go run cmd/cve/generate.go` locally on my branch. Then I switch my test file tmp dir to be the directory the json files are outputted to via `NVD_TEST_VULNDB_DIR`

Any thoughts on how to QA this better?

Reply:
Yeah, vulns dir + vuln processing run would do the trick. Just need to pull the affected version of PowerShell and make sure it shows the correct vulns.

Reply:
Thought it would be harder to find rc1
https://github.com/PowerShell/PowerShell/releases/tag/v7.5.0-rc.1